File: javascript/src/models/event_type_patch.ts
// this file is @generated

export interface EventTypePatch {
  archived?: boolean;
  deprecated?: boolean;
  description?: string;
  featureFlag?: string | null;
  /** The event type group's name */
  groupName?: string | null;
  schemas?: any | null;
}

export namespace EventTypePatchUtil {
  export function _fromJsonObject(object: any): EventTypePatch {
    return {
      archived: object["archived"],
      deprecated: object["deprecated"],
      description: object["description"],
      featureFlag: object["featureFlag"],
      groupName: object["groupName"],
      schemas: object["schemas"],
    };
  }

  export function _toJsonObject(self: EventTypePatch): any {
    return {
      archived: self.archived,
      deprecated: self.deprecated,
      description: self.description,
      featureFlag: self.featureFlag,
      groupName: self.groupName,
      schemas: self.schemas,
    };
  }
}

File: kotlin/lib/src/main/kotlin/models/EventTypePatch.kt
// This file is @generated
package com.svix.kotlin.models

import com.svix.kotlin.MaybeUnset
import kotlinx.serialization.Serializable
import kotlinx.serialization.json.JsonObject

@Serializable
data class EventTypePatch(
    val archived: Boolean? = null,
    val deprecated: Boolean? = null,
    val description: String? = null,
    val featureFlag: MaybeUnset<String> = MaybeUnset.Undefined,
    val groupName: MaybeUnset<String> = MaybeUnset.Undefined,
    val schemas: MaybeUnset<JsonObject> = MaybeUnset.Undefined,
)
